modules/tensor_mechanics/test/include/userobjects/TestSubblockIndexProvider.h

```diff
@@ -36,4 +35,8 @@ class TestSubblockIndexProvider : public SubblockIndexProvider
    * The max index of subblock.
    */
   virtual unsigned int getMaxSubblockIndex() const override;
+
+protected:
+  /// A Userobject that carries the subblock ID for all elements
+  const SubblockIndexProvider * _subblock_id_provider;
```

Comment:
I don't understand why you add this. This object _is_ the SubblockIndexProvider, it doesn't couple to one